internal/govulncheckshim/client.go

```diff
@@ -0,0 +1,110 @@
+package govulncheckshim
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/google/osv-scanner/pkg/models"
+	"golang.org/x/vuln/client"
+	gvcOSV "golang.org/x/vuln/osv"
+)
+
+type localSource struct {
+	vulnList         []models.Vulnerability
+	vulnsByID        map[string]*models.Vulnerability
+	vulnsByAlias     map[string][]*models.Vulnerability
+	vulnsByModule    map[string][]*models.Vulnerability
+	lastModifiedTime time.Time
+	client.Client
+}
+
+func newClient(vulns []models.Vulnerability) *localSource {
+	client := localSource{
+		vulnList:         vulns,
+		vulnsByID:        make(map[string]*models.Vulnerability),
+		vulnsByAlias:     make(map[string][]*models.Vulnerability),
+		vulnsByModule:    make(map[string][]*models.Vulnerability),
+		lastModifiedTime: time.Unix(0, 0),
+	}
+	for idx := range vulns {
+		// Iterate on reference to avoid copying entire data structure
+		v := &client.vulnList[idx]
+		client.vulnsByID[v.ID] = v
+		for _, alias := range v.Aliases {
+			client.vulnsByAlias[alias] = append(client.vulnsByAlias[alias], v)
+		}
+		for _, affected := range v.Affected {
+			client.vulnsByModule[affected.Package.Name] = append(client.vulnsByModule[affected.Package.Name], v)
+		}
+		if client.lastModifiedTime.Before(v.Modified) {
+			client.lastModifiedTime = v.Modified
+		}
+	}
+
+	return &client
+}
+
+func convertToGvcOSV(osv models.Vulnerability) gvcOSV.Entry {
```

Comment:
I'd consider turning panics here into an error and propagate them through their callees. But this also depends on how you decide the scanner should communicate failures to users.

Reply:
Updated to return an error instead of panicing, though I'm not sure what the callees or user can do if the two types differ, I believe this should be caught during development if the types do change to be incompatible. 

Still this  does make it clearer to the user and keeps the non analysis part running.